### ISSUE TITLE
TensorFlow API 之 nn

### ISSUE THREAD

**Issue by C4droid** [C4droid/Blog#40]
# 激活函数
在神经网络中，我们有很多的非线性函数来作为激活函数，比如连续的平滑非线性函数（sigmoid，tanh和softplus），连续但不平滑的非线性函数（relu，relu6和relu_x）和随机正则化函数（dropout）。

所有的激活函数都是单独应用在每个元素上面的，并且输出张量的维度和输入张量的维度一样。

## tf.nn.relu(features, name = None)
将矩阵中每行的负数置0

## tf.nn.bias_add(value, bias, name = None)
这个函数的作用是将偏差项 bias 加到 value 上面。其中 bias 必须是一维的

## tf.nn.conv2d(input,filter,strides,padding,use_cudnn_on_gpu=None, data_format=None,name=None)

- strides: A list of ints.一维，长度4
在input上切片采样时，每个方向上的滑窗步长，必须和format指定的维度同阶
strides = [1, 1, 1, 1] 表示卷积核对每个像素点进行卷积，即在二维屏幕上面，两个轴方向的步长都是1。strides = [1, 2, 2, 1] 表示每隔一个像素点进行卷积，两个轴方向的步长都是2

- data_format: An optional string from: "NHWC", "NCHW"， 默认为"NHWC"。
```python
input_layer = tf.reshape(features, [batch, height, width, channel])
# H和W，height和width很好理解，指的是图片长宽各多少像素
# N，batch则是指图片的数量，一批数据中有多少张图片
# C，channel，通道，图片的类型。为1，则是灰度图；为3则是rgb彩图
```

- padding:一个字符串，取值为 SAME 或者 VALID
SAME会自动填充空白
VALID会舍弃多余地、不足以覆盖的数据

**Comment by C4droid** [comment]
## 激活函数
在神经网络中，我们有很多的非线性函数来作为激活函数，比如连续的平滑非线性函数（sigmoid，tanh和softplus），连续但不平滑的非线性函数（relu，relu6和relu_x）和随机正则化函数（dropout）。

所有的激活函数都是单独应用在每个元素上面的，并且输出张量的维度和输入张量的维度一样。

**tf.nn.relu(features, name = None)**
将矩阵中每行的负数置0

**Comment by C4droid** [comment]
## tf.nn.bias_add(value, bias, name = None)
这个函数的作用是将偏差项 bias 加到 value 上面。其中 bias 必须是一维的

**Comment by C4droid** [comment]
**tf.nn.conv2d(input,filter,strides,padding,use_cudnn_on_gpu=None, data_format=None,name=None)**

- strides: A list of ints.一维，长度4
在input上切片采样时，每个方向上的滑窗步长，必须和format指定的维度同阶
strides = [1, 1, 1, 1] 表示卷积核对每个像素点进行卷积，即在二维屏幕上面，两个轴方向的步长都是1。strides = [1, 2, 2, 1] 表示每隔一个像素点进行卷积，两个轴方向的步长都是2

- data_format: An optional string from: "NHWC", "NCHW"， 默认为"NHWC"。
```python
input_layer = tf.reshape(features, [batch, height, width, channel])
# H和W，height和width很好理解，指的是图片长宽各多少像素
# N，batch则是指图片的数量，一批数据中有多少张图片
# C，channel，通道，图片的类型。为1，则是灰度图；为3则是rgb彩图
```

- padding:一个字符串，取值为 SAME 或者 VALID
SAME会自动填充空白
VALID会舍弃多余地、不足以覆盖的数据

**Comment by C4droid** [comment]
### tf.nn.dropout()
tf.nn.dropout(x, keep_prob, noise_shape = None, seed = None, name = None)

一个神经元将以概率keep_prob决定是否放电，如果不放电，那么该神经元的输出将是0，如果该神经元放电，那么该神经元的输出值将被放大到原来的1/keep_prob倍。

比如，神经元的值为[1, 2]，keep_prob的值是0.5，并且是第一个神经元是放电的，第二个神经元不放电，那么神经元输出的结果是[2, 0]

- x: 一个Tensor。
- keep_prob: 一个 Python 的 float 类型。表示元素是否放电的概率。
- noise_shape: 一个一维的Tensor，数据类型是int32。代表元素是否独立的标志。
- seed: 一个Python的整数类型。设置随机种子。
- name: （可选）为这个操作取一个名字。
